### ISSUE TITLE
ASI Loader with a DX7 game (Hitman Codename 47)

### ISSUE THREAD

**Issue by vrubayka** [ThirteenAG/Ultimate-ASI-Loader#52]
I want to use the loader with a DX7 game and also use [DSOAL](http://vaporeon.io/hosted/dsoal-builds/dsoal-latest.7z) (`dsound.dll`). How do I rename the loader `.dll`? DDraw.dll isn't an option since I'm also using dgVoodoo2. 

Everything I tried (dinput, etc.) hasn't been hooked. I tried renaming the DSOAL `.dll` into an `.asi` and putting all of its files in a properly set-up `scripts` folder, but DSOAL didn't work. The scripts folder contains a `global.ini` and I'm already hooking a widescreen fix .asi so it should be working.

**Comment by Kappa971** [comment]
I had opened a problem a while ago: https://github.com/ThirteenAG/WidescreenFixesPack/issues/1024

**Comment by ThirteenAG** [comment]
Can you upload exe of this game?

**Comment by Kappa971** [comment]
Sure, this is version 1.92 (the latest): [Hitman.zip](https://github.com/ThirteenAG/Ultimate-ASI-Loader/files/8054389/Hitman.zip)

**Comment by Kappa971** [comment]
This game also has an OpenGL render that doesn't suffer from the same problems as DirectX7 with Windows 10/11. Unfortunately the Widescreen fix only works with DirectX7.

**Comment by vrubayka** [comment]
[Hitman.zip](https://github.com/ThirteenAG/Ultimate-ASI-Loader/files/8055415/Hitman.zip)
Kappa971's .exe doesn't work for me, so here's mine. 
Basically my issue is that I'm using D3DImm.dll and DDraw.dll from dgVoodoo2 and dsound.dll from DSOAL. Is there any way I can rename the ASI loader, so that it works with the game?

**Comment by Kappa971** [comment]
The game works perfectly as you can see, maybe your game is another version:
![Hitman  Codename 47 Screenshot 2022 02 13 - 13 05 17 93](https://user-images.githubusercontent.com/62349018/153752299-8cc405b5-3443-4ee8-bf24-a24475c59e96.png)

> Basically my issue is that I'm using D3DImm.dll and DDraw.dll from dgVoodoo2 and dsound.dll from DSOAL. Is there any way I can rename the ASI loader, so that it works with the game?

I guess the developer needs to make some changes to Ultimate ASI Loader.
A Widescreen fix for OpenGL rendering would perhaps be better (but I guess more complex) instead of using the existing Widescreen fix in DirectX7.

**Comment by ThirteenAG** [comment]
I guess I'd need a dump of exe instead.

**Comment by Kappa971** [comment]
> I guess I'd need a dump of exe instead.

What should I do? I never did it.

**Comment by ThirteenAG** [comment]
Use something like PETools, select process, then dump full, save.

**Comment by vrubayka** [comment]
[Dumped.zip](https://github.com/ThirteenAG/Ultimate-ASI-Loader/files/8056041/Dumped.zip)

**Comment by ThirteenAG** [comment]
Unfortunately those don't have original imports table, so I checked with debugger, I think msacm32.dll, version.dll or winmm.dll should still work.

**Comment by Kappa971** [comment]
> I think msacm32.dll, version.dll or winmm.dll should still work.

I think I have already tried to rename the Ultimate ASI Loader dll in all the names present in the description on the first page, only ddraw.dll (and dsound.dll even if I have not tried it personally, as that dll is already in use from DSOAL) seem to load the Widescreen fix. Perhaps @vrubayka can confirm it.

> Unfortunately those don't have original imports table

Can I help in any way?

**Comment by ThirteenAG** [comment]
> I think I have already tried to rename the Ultimate ASI Loader dll in all the names present in the description on the first page, only ddraw.dll (and dsound.dll even if I have not tried it personally, as that dll is already in use from DSOAL) seem to load the Widescreen fix. Maybe @vrubayka can confirm it.

Then I guess there's nothing I can do?

**Comment by Kappa971** [comment]
> Then I guess there's nothing I can do?

I guess not, unless you create a Widescreen fix for OpenGL rendering if possible. If it isn't possible, I think that this issue and mine opened previously, can be closed.
Anyway, thank you for your patience.

**Comment by vrubayka** [comment]
Yeah only dsound and ddraw get hooked sadly. Anyway hopefully somebody will get this to work maybe with a different wrapper or something.

**Comment by Kappa971** [comment]
> Yeah only dsound and ddraw get hooked sadly. Anyway hopefully somebody will get this to work maybe with a different wrapper or something.

A Widescreen fix for OpenGL would be perfect as:
1. You can turn on anti-aliasing without corrupting the text.
2. It is more compatible with new versions of Windows and allows you to activate an fps limiter (in case you have a Freesync/G-Sync display with high refresh rate).
But I think that if it hasn't been done so far, maybe it isn't possible (or too complex).
